### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**

### CODE REVIEW

File: src/MyEventEmitter.js [hunk 1]
```diff
@@ -1,12 +1,80 @@
...
+      this.events.set(event, [...thisEvent, callbackWrapper])
+    } else {
+      this.events.set(event, [callbackWrapper])
+    }
+    return this
+  }
+  off(event, callback) {
+    const eventList = this.events
+      .get(event)
+      .filter((c) => c + "" !== callback + "")
```
Comment (sergii-nosachenko) [inline src/MyEventEmitter.js:33]: Why do you convert and compare functions as strings?
Remind me how we store objects in memory? Why do we call real event.off() method with the same function (not copy or else)?

File: src/MyEventEmitter.js [hunk 2]
```diff
@@ -1,12 +1,80 @@
...
+    } else {
+      this.events.set(event, [callback])
+    }
+    return this
+  }
+  prependOnceListener(event, callback) {
+    const callbackWrapper = () => {
+      callback()
+      this.off(event, callbackWrapper)
+    }
```
Comment (sergii-nosachenko) [inline src/MyEventEmitter.js:59]: Create class method for this

File: src/MyEventEmitter.js [hunk 3]
```diff
@@ -1,12 +1,80 @@
...
+      this.events.delete(event)
+      this.events.set(event, [callbackWrapper, ...thisEvent])
+    } else {
+      this.events.set(event, [callbackWrapper])
+    }
+    return this
+  }
+  removeAllListeners() {
+    this.events.clear()
+    return this
```
Comment (sergii-nosachenko) [inline src/MyEventEmitter.js:72]:
```suggestion

    return this;
```
Overall code style is bad, you can improve a lot. First of all - add linter to this task and make your code more pretty and readable.

File: src/MyEventEmitter.js [hunk 4]
```diff
@@ -1,12 +1,80 @@
 'use strict';
 
 export class MyEventEmitter {
```
Comment (sergii-nosachenko) [inline src/MyEventEmitter.js:3]: To pass linter check use module.exports instead

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by Anna-Tkachenko — APPROVED** [review]
Good job! ⭐️

**Pull request closed without merging**